office365_connector.py

```diff
@@ -3303,6 +2981,84 @@ def _handle_update_email(self, param):
+            action_result, endpoint, data=json.dumps({"moveToJunk": move_to_junk_folder}), method="post", beta=True
+        )
+
+        if phantom.is_fail(ret_val):
+            return action_result.set_status(phantom.APP_ERROR, f"Moving email  with id: {message} to junk folder failed")
+
+        action_result.add_data(response)
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_unblock_sender(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        message = param["message_id"]
+        user = param["user_id"]
+        move_to_inbox = param.get("move_to_inbox", False)
+
+        endpoint = f"/users/{user}/messages/{message}/markAsNotJunk"
+        self.save_progress(f"endpoint {endpoint}")
+
+        ret_val, response = self._make_rest_call_helper(
+            action_result, endpoint, data=json.dumps({"moveToInbox": move_to_inbox}), method="post", beta=True
+        )
+
+        if phantom.is_fail(ret_val):
+            return action_result.set_status(phantom.APP_ERROR, f"Moving email  with id: {message} to inbox folder failed")
+
+        action_result.add_data(response)
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_resolve_name(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # id or userPrincipalName
+        email = param["email"]
+
+        endpoint = f"/users?$filter=startswith(displayName,'{email}') or startswith(mail,'{email}')"
+        ret_val, responses = self._make_rest_call_helper(action_result, endpoint)
+
+        if phantom.is_fail(ret_val):
+            return action_result.set_status(phantom.APP_ERROR, "Got invalid ret val")
+
+        for response in responses.get("value"):
+
+            user_id = response.get("id")
+
+            endpoint_other_mails = f"/users/{user_id}?$select=mailNickname,proxyAddresses,otherMails"
+            endpoint_other_address = f"/users/{user_id}?$select=city,state,street,postalCode"
+            endpoint_mailbox = f"/users/{user_id}/mailboxSettings/userPurpose"
+
+            ret_val_proxy, response_proxy = self._make_rest_call_helper(action_result, endpoint_other_mails)
+            ret_val_address, response_address = self._make_rest_call_helper(action_result, endpoint_other_address)
+            ret_val_mailbox, response_mailbox = self._make_rest_call_helper(action_result, endpoint_mailbox)
+
+            self.save_progress(f"{ret_val} Got response {response}")
+            self.save_progress(f"{ret_val_proxy} Got response other mails {response_proxy}")
+            self.save_progress(f"{ret_val_address} Got response address {response_address}")
+            self.save_progress(f"{ret_val_mailbox} Got response mailbox {response_mailbox}")
```

Comment:
should we print output of everything? Output may be large depending on response count, maybe debug_print or other would be better?

Reply:
I have shortened the outputs, I believe due to nature of msgraph asset it is easier to debug and maintain action while using save_progress.